src/main/java/frc/trigon/robot/poseestimation/poseestimator/PoseEstimator.java

```diff
@@ -113,41 +137,126 @@ private void logTargetPath() {
         PathPlannerLogging.setLogTargetPoseCallback((pose) -> Logger.recordOutput("TargetPPPose", pose));
     }
 
+    /**
+     * Sets the estimated pose from the odometry at the given timestamp.
+     *
+     * @param swerveModulePositions the positions of each swerve module
+     * @param gyroHeading           the heading of the gyro
+     * @param timestamp             the timestamp of the odometry observation
+     */
+    private void addOdometryObservation(SwerveModulePosition[] swerveModulePositions, Rotation2d gyroHeading, double timestamp) {
```

Comment:
addOdometryobservation still isn't a good name. What's an "observation" think of smth better

Reply:
I have no problem with `observation`. If you have any better ideas, I'd be glad to hear them